tests/src/test/scala/whisk/core/containerpool/test/ContainerProxyTests.scala

```diff
@@ -318,6 +318,59 @@ class ContainerProxyTests extends TestKit(ActorSystem("ContainerProxys"))
         }
     }
 
+    it should "resend the job to the parent if resuming a container fails" in within(timeout) {
+        val container = new TestContainer {
+            override def resume()(implicit transid: TransactionId) = {
+                resumeCount += 1
+                Future.failed(new RuntimeException())
+            }
+        }
+        def factory(tid: TransactionId, name: String, image: ImageName, pull: Boolean, memoryLimit: ByteSize) = Future.successful(container)
+
+        val machine = childActorOf(ContainerProxy.props(factory, ack, store))
+        registerCallback(machine)
+        run(machine, Uninitialized) // first run an activation
+        timeout(machine) // times out Ready state so container suspends
+        expectPause(machine)
+
+        val runMessage = Run(action, message)
+        machine ! runMessage
+        expectMsg(Transition(machine, Paused, Running))
+        expectMsg(ContainerRemoved)
+        expectMsg(Transition(machine, Running, Removing))
+        expectMsg(runMessage)
+
+        awaitAssert {
+            container.runCount shouldBe 1
+            container.suspendCount shouldBe 1
+            container.resumeCount shouldBe 1
+            container.destroyCount shouldBe 1
+        }
+    }
+
+    it should "remove the container if suspend fails" in within(timeout) {
+        val container = new TestContainer {
+            override def suspend()(implicit transid: TransactionId) = {
+                suspendCount += 1
+                Future.failed(new RuntimeException())
+            }
+        }
+        def factory(tid: TransactionId, name: String, image: ImageName, pull: Boolean, memoryLimit: ByteSize) = Future.successful(container)
+
+        val machine = childActorOf(ContainerProxy.props(factory, ack, store))
+        registerCallback(machine)
+        run(machine, Uninitialized)
+        timeout(machine) // times out Ready state so container suspends
+        expectMsg(Transition(machine, Ready, Pausing))
+        expectMsg(ContainerRemoved)
+        expectMsg(Transition(machine, Pausing, Removing))
+
+        awaitAssert {
+            container.suspendCount shouldBe 1
+            container.destroyCount shouldBe 1
+        }
```

Comment:
Thanks for adding these. If docker gets sick and removing itself fails is there anything else that can be done? I'm thinking of some of the cases we've encountered in the past iirc.